File: frontend/src/api/staff.ts
import { api } from './client'

export type OrganizationRole = 'owner' | 'admin' | 'manager'
export type ProjectMemberRole = 'project_admin' | 'content_manager' | 'webinar_moderator' | 'support' | 'analyst'

export interface StaffProjectAccess {
  project_id: number
  project_name: string
  project_color: string
  role: ProjectMemberRole
  created_at: string
}

export interface StaffUser {
  id: number
  organization_id: number | null
  email: string
  full_name: string | null
  role: OrganizationRole
  is_active: boolean
  created_at: string
  projects: StaffProjectAccess[]
}

export interface ProjectMember {
  id: number
  project_id: number
  user_id: number
  email: string
  full_name: string | null
  organization_role: OrganizationRole
  role: ProjectMemberRole
  is_active: boolean
  created_at: string
}

export const staffApi = {
  list: () => api.get<StaffUser[]>('/admin/staff/').then((r) => r.data),
  create: (data: {
    email: string
    full_name?: string
    temp_password?: string
    organization_role: OrganizationRole
    project_id: number
    project_role: ProjectMemberRole
  }) => api.post<StaffUser>('/admin/staff/', data).then((r) => r.data),
  update: (userId: number, data: { full_name?: string; organization_role?: OrganizationRole; is_active?: boolean }) =>
    api.patch<StaffUser>(`/admin/staff/users/${userId}`, data).then((r) => r.data),
  projectMembers: (projectId: number) =>
    api.get<ProjectMember[]>(`/admin/staff/projects/${projectId}/members`).then((r) => r.data),
  addProjectMember: (projectId: number, data: {
    email: string
    full_name?: string
    temp_password?: string
    organization_role: OrganizationRole
    role: ProjectMemberRole
  }) => api.post<ProjectMember>(`/admin/staff/projects/${projectId}/members`, data).then((r) => r.data),
  updateProjectMember: (projectId: number, userId: number, role: ProjectMemberRole) =>
    api.patch<ProjectMember>(`/admin/staff/projects/${projectId}/members/${userId}`, { role }).then((r) => r.data),
  removeProjectMember: (projectId: number, userId: number) =>
    api.delete(`/admin/staff/projects/${projectId}/members/${userId}`),
}

File: backend/app/schemas/staff.py
from datetime import datetime
from pydantic import BaseModel, EmailStr

from app.models.project_member import ProjectMemberRole
from app.models.user import UserRole


class StaffProjectAccessOut(BaseModel):
    project_id: int
    project_name: str
    project_color: str
    role: ProjectMemberRole
    created_at: datetime


class StaffUserOut(BaseModel):
    id: int
    organization_id: int | None
    email: str
    full_name: str | None
    role: UserRole
    is_active: bool
    created_at: datetime
    projects: list[StaffProjectAccessOut] = []

    model_config = {"from_attributes": True}


class StaffCreate(BaseModel):
    email: EmailStr
    full_name: str | None = None
    temp_password: str | None = None
    organization_role: UserRole = UserRole.manager
    project_id: int
    project_role: ProjectMemberRole


class StaffUpdate(BaseModel):
    full_name: str | None = None
    organization_role: UserRole | None = None
    is_active: bool | None = None


class ProjectMemberCreate(BaseModel):
    email: EmailStr
    full_name: str | None = None
    temp_password: str | None = None
    organization_role: UserRole = UserRole.manager
    role: ProjectMemberRole


class ProjectMemberUpdate(BaseModel):
    role: ProjectMemberRole


class ProjectMemberOut(BaseModel):
    id: int
    project_id: int
    user_id: int
    email: str
    full_name: str | None
    organization_role: UserRole
    role: ProjectMemberRole
    is_active: bool
    created_at: datetime
